### PR TITLE
[IMP] project_{*}: propagate project field on MTO replenishment

### PR DESCRIPTION
**Pull request by mohammadhossam**
This PR ensures that the project field is propagated in different MTO scenarios. This commit specifically focuses on: MO -> Child MO, MO -> PO, SO -> PO scenarios.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale_project/models/sale_order_line.py [hunk 1]
```diff
@@ -416,3 +416,9 @@ def _get_action_per_item(self):
             :returns: Dict containing id of SOL as key and the action as value
         """
         return {}
+
+    def _prepare_procurement_values(self, group_id=False):
+        values = super()._prepare_procurement_values(group_id=group_id)
+        if self.project_id:
+            values['project_id'] = self.order_id.project_id.id
```
Comment (Whenrow) [inline addons/sale_project/models/sale_order_line.py:423]: sale.order.line also have `project_id` which seems more precise than the SO one

Comment (xavierbol) [reply addons/sale_project/models/sale_order_line.py:423]: @Whenrow Actually, the `project_id` field in SOL is more used to store the project generated by the SOL, so, I think it is better to use the one defined in SO as we did here. 🙂

Comment (Whenrow) [reply addons/sale_project/models/sale_order_line.py:423]: @xavierbol ok I guess this `project_id` is only relevant with service product on SO line

Comment (xavierbol) [reply addons/sale_project/models/sale_order_line.py:423]: @Whenrow yep exactly. 🙂

**Pull request closed without merging**